packages/test/test-end-to-end-tests/src/test/opPerfTelemetry.spec.ts

```diff
@@ -0,0 +1,145 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import {
+	ITestObjectProvider,
+	ITestContainerConfig,
+	DataObjectFactoryType,
+	ITestFluidObject,
+	createTestConfigProvider,
+} from "@fluidframework/test-utils";
+import { type SharedString } from "@fluidframework/sequence";
+import { describeCompat, itExpects } from "@fluid-private/test-version-utils";
+import { ILoaderProps } from "@fluidframework/container-loader";
+
+/**
+ * Regression tests for issue where clients loading a container with a write connection (vs the read connection
+ * that became the default) would never generate OpRoundtripTime telemetry.
+ */
+describeCompat(
+	"All clients generate OpRoundtripTime telemetry",
+	// TODO: this should be "2.0.0-rc.2.0.0" (I think) once the bugs with describeCompat are fixed and versions are used correctly
+	"NoCompat",
+	(getTestObjectProvider, apis) => {
+		const { SharedString } = apis.dds;
+
+		const customConfigProvider = createTestConfigProvider();
+		customConfigProvider.set("Fluid.Container.ForceWriteConnection", true);
+		const loaderPropsThatForceWriteConnection: Partial<ILoaderProps> = {
+			configProvider: customConfigProvider,
+		};
+
+		const stringId = "stringKey";
+		const configWithReadConnection: ITestContainerConfig = {
+			fluidDataObjectType: DataObjectFactoryType.Test,
+			registry: [[stringId, SharedString.getFactory()]],
+		};
+		const configWithWriteConnection = {
+			...configWithReadConnection,
+			loaderProps: loaderPropsThatForceWriteConnection,
+		};
+
+		let provider: ITestObjectProvider;
+		beforeEach("getTestObjectProvider", () => {
+			provider = getTestObjectProvider();
+		});
+
+		itExpects(
+			"Create with read connection and load with read connection",
+			[
+				// Expect one OpRoundtripTime event from each client
+				{ eventName: "fluid:telemetry:OpRoundtripTime", clientType: "interactive" },
```

Comment:
@alexvy86 -- I noticed that these tests pass if you have _not enough_ of these events listed in the `itExpects` list, because they're logged as `generic` since the timing is not slower than 5s.  So I would suggest using MockLogger to assert that you get exactly 2 and no more, rather than using `itExpects`.

Reply:
Hmmm not opposed to using MockLogger, but I don't get the reason for the failure. `itExpects()` doesn't check that the list of generated logs matches the provided one exactly?

Reply:
> itExpects() doesn't check that the list of generated logs matches the provided one exactly?

It does not.

I refactored the tests to be more robust in https://github.com/microsoft/FluidFramework/pull/22361. Using MockLogger naively had its disadvantages too; the actual (comprehensive) list of events emitted is pretty big and unrelated to this test, so it feels icky and brittle to try to do a strict exact match. Since these tests are trying to validate that the event is emitted by a container both when it's created and when it's loaded, they now check for one instance of the event after creating (and clear the events seen by MockLogger), and a separate instance after loading. This should prevent a false positive if the creation flow happened to emit the log twice while the load one didn't emit it at all.